llm_unlearn_ucl/unlearn_harm.py

```diff
@@ -372,27 +405,32 @@ def main(args) -> None:
     model.train()
 
     # Reference model for computing KL.
+    if accelerator.is_local_main_process:
+        print(f"Loading model {args.model_name} for reference ('fully learned')...")
 
-    print(f"Loading model {args.model_name} for reference ('fully learned')...")
     if args.use_quantized:
         # Uncomment for quantized
         pretrained_model = AutoModelForCausalLM.from_pretrained(
             args.model_name,
             cache_dir=args.cache_dir,
             load_in_8bit=True,
-            torch_dtype=torch.float32,
+            torch_dtype=torch.bfloat16,
+            trust_remote_code=True,
         )
     else:
         pretrained_model = AutoModelForCausalLM.from_pretrained(
             args.model_name,
             cache_dir=args.cache_dir,
             torch_dtype=torch.bfloat16,
```

Comment:
For visibility: @Willmish's code removes bf16 from the python code and defines it in the deepspeed config.